models/openai_model.py

```diff
@@ -776,20 +781,24 @@ async def send_request(
                 f"Prompt must be greater than {self.prompt_min_length} characters, it is currently: {len(prompt)} characters"
             )
 
+        if not max_tokens_override:
+            if model:
+                max_tokens_override = Models.get_max_tokens(model) - tokens
+
         print(f"The prompt about to be sent is {prompt}")
         print(
             f"Overrides -> temp:{temp_override}, top_p:{top_p_override} frequency:{frequency_penalty_override}, presence:{presence_penalty_override}"
         )
 
-        async with aiohttp.ClientSession(raise_for_status=True) as session:
+        async with aiohttp.ClientSession(raise_for_status=False) as session:

```

Comment:
Why did you make this not raise it anymore? Doesn't ValueError not catch other possible http exception cases? ValueError can only be possible if the API actually responds and we try to index the response json, but there are other client response errors that can occur right?